tests/phpunit/tests/test-create-delete-languages.php

```diff
@@ -171,4 +171,43 @@ public function test_language_properties_in_transient() {
 		$languages = get_transient( 'pll_languages_list' );
 		$this->assertEqualSets( $properties, array_keys( reset( $languages ) ) );
 	}
+
+	/**
+	 * @doesNotPerformAssertions
+	 */
+	public function test_create_secondary_language_with_yoast() {
+		// first language
+		$args = array(
+			'name'       => 'English',
+			'slug'       => 'en',
+			'locale'     => 'en_US',
+			'rtl'        => 0,
+			'flag'       => 'us',
+			'term_group' => 2,
+		);
+		self::$model->add_language( $args );
+
+		$links_model     = self::$model->get_links_model();
```

Comment:
I suspect that the test doesn't reproduce the issue. Indeed, the issue is about:
`Trying to get property 'slug' of non-object in /include/links-directory.php on line 146`
However, by default the links model is an instance of `PLL_Links_Default`.
Now, if you look at https://github.com/polylang/polylang/blob/3.1.4/include/links-directory.php#L146 and https://github.com/polylang/polylang/blob/3.1.4/include/links-model.php#L156, you'll see that `PLL_Links_Directory->home_url()` always accesses to `$lang->slug` while it's not true for `PLL_Links_Default->home_url()`.
To reproduce the issue, you need to either make sure your links model is an instance of `PLL_Links_Directory` or make sure  that `$options['hide_default']` is true. The second is probably easier to setup.

Reply:
Indeed this test focused more on the second second of the undefined index.
It should test both now.